Settings page refinements

pepsipotty opened the pull request.

pepsipotty committed:
just some testing

pepsipotty committed:
add view model to better handle transient data persistence

pepsipotty committed:
added AnimatedBorderCard

pepsipotty committed:
remove unused imports

pepsipotty committed:
remove unused function

pepsipotty committed:
remove unused lines

pepsipotty committed:
use intValue instead of value to reduce autoboxing overhead

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/MainScreen.kt:
```diff
@@ -84,8 +88,22 @@ fun MainScreen(
+                                snackbarHostState.showSnackbar(
+                                    "API Key is required to proceed.",
+                                    duration = SnackbarDuration.Short
+                                )
```
nice caution and attention to detail!

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/NamingScreen.kt:
```diff
@@ -163,4 +173,47 @@ fun NamingScreen(drawerState: DrawerState, scope: CoroutineScope, base64EncodedP
+}
+
+@Composable
+fun TextInputField(tokenValue: MutableState<TextFieldValue>, labelText: String) {
```
is this being _copied_ from the other file to here? I don't see it being used here. I'm confused

BareBeaverBat replied:
if we're going to have one composable called TextInputField in one file and another with the same name in the other file, could you please make each one private (so it's only visible in its own file)?

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/Settings.kt:
```diff
@@ -227,9 +221,9 @@ fun TextInputField(tokenValue: MutableState<TextFieldValue>, labelText: String)
-        value = tokenValue.value,
-        onValueChange = {
-            tokenValue.value = it
+        value = tokenValueInput.value?.let { it } ?: tokenValueState,
```
```suggestion
        value = tokenValueInput.value ?: tokenValueState,
```
per ide, this should be equivalent

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/Settings.kt:
```diff
@@ -329,5 +344,54 @@ fun ModelCard(id: Int, title: String, description: String, selectedModel: Mutabl
 }
 
+@Composable
+fun AnimatedBorderCard(
```
this is sweet :D

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/Settings.kt:
```diff
@@ -298,7 +308,7 @@ fun ModelCard(id: Int, title: String, description: String, selectedModel: Mutabl
                     //Description
                     Text(
-                        color = if (selectedModel.value == id) MaterialTheme.colorScheme.primary else Color.Gray,
+                        color = txtColor,
```
yay, DRY!

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/model/SettingsViewModel.kt:
```diff
@@ -0,0 +1,48 @@
+    val tokenValue: State<TextFieldValue> = _tokenValue
+
+    private val _selectedModel = mutableIntStateOf(0)
+    val selectedModel: State<Int> = _selectedModel
```
even after going through the whole PR, I'm still finding it confusing that we have State/MutableState objects in the ViewModel. Since you also have MutableState objects in the SettingsScreen composable to keep the ui responsively updating to user input, why not have these viewmodel instance variables be simple strings and ints?

BareBeaverBat reviewed (changes requested):
Looks really sweet, and works in my testing.
A couple code decisions I'm questioning- I'd be happy if you explained your reasoning and convinced me that my skepticism was unwarranted
I'll ofc resolve the positive comments during the second round of review

pepsipotty committed:
Refactor: Remove redundant .let keyword.

Addresses feedback from: https://github.com/BareBeaverBat/CSE5236-SU24-DungeonMapDigitizer/pull/9/files/dadf5a872172d353306858c0427878989a7d0215#r1650077277

"per ide, this should be equivalent"

pepsipotty committed:
Refactor: Change _tokenValue and _selectedModel to hold simple string and int values respectively

Addresses feedback from: https://github.com/BareBeaverBat/CSE5236-SU24-DungeonMapDigitizer/pull/9/files/dadf5a872172d353306858c0427878989a7d0215#r1650078745

"Even after going through the whole PR, I'm still finding it confusing that we have State/MutableState objects in the ViewModel. Since you also have MutableState objects in the SettingsScreen composable to keep the ui responsively updating to user input, why not have these viewmodel instance variables be simple strings and ints?"

_tokenValue and _selectedModel are now simple strings and ints respectively

pepsipotty committed:
Remove unused imports and some autoboxing optimizations

pepsipotty committed:
Remove excess whitespace

BareBeaverBat commented on app/src/main/java/com/example/dungeontest/Settings.kt:
```diff
@@ -206,7 +198,7 @@ fun SettingsScreen(
 }
 
 @Composable
-fun TextInputField(tokenValue: MutableState<TextFieldValue>, labelText: String) {
```
if we're going to have one composable called TextInputField in one file and another with the same name in the other file, could you please make each one private (so it's only visible in its own file)?

BareBeaverBat reviewed (changes requested):
1-2 small changes requested before approval

pepsipotty committed:
specify TextInputField as private functions to avoid reference ambiguity

BareBeaverBat reviewed (approved):
thanks for working with me
